### PR TITLE
refactor(internal): Doing a TODO for UpgradeDisconnectionShape

### PR DESCRIPTION
**Pull request by erights**
closes: #XXXX
refs: #8742 #7401 

## Description

Happened to run across this TODO from #7401 while reviewing #8742. Looking, at the #7401 review, I see at https://github.com/Agoric/agoric-sdk/pull/7401/files#diff-0bf8845b178624f94f798671028604656a8848e55e1f41784fa4b37b5236ee96

> My first attempt didn't quite work, so I'm deferring this until we've got a little more time to spend on it.

So I thought I'd give it a trivial try and see what CI has to say about it.
As a result: It seems the problem is that `isUpgradeDisconnection` was sometimes applied to non-frozen non-UpgradeDisconnection objects, such as errors thrown by the platform or by a raw `throw *Error(...);`. So added an early `isFrozen` conjunct.

Also
- since the UpgradeDisconnection is not and error, but is to be used as a promise-rejection-reason, changed terminology from `err` to `reason`.
- given the rest of the API, IMO the type should be named `UpgradeDisconnection`. So I did that rename.

### CODE REVIEW

File: packages/SwingSet/src/types-external.js [hunk 1]
```diff
@@ -60,7 +60,7 @@ export {};
  * @typedef { import('@agoric/swingset-liveslots').Message } Message
  *
  * @typedef { 'none' | 'ignore' | 'logAlways' | 'logFailure' | 'panic' } ResolutionPolicy
- * @typedef {import('@agoric/internal/src/upgrade-api.js').DisconnectionObject} DisconnectionObject
+ * @typedef {import('@agoric/internal/src/upgrade-api.js').UpgradeDisconnection} UpgradeDisconnection
```
Comment (erights) [inline packages/SwingSet/src/types-external.js]: It does not seem anyone actually uses this typedef. Should I delete it instead?

Comment (gibson042) [reply packages/SwingSet/src/types-external.js]: I'm fine with this either way.

File: packages/internal/src/upgrade-api.js [hunk 2]
```diff
@@ -1,18 +1,34 @@
...
+
+const { isFrozen } = Object;
 
 /**
- * @typedef {{ name: string, upgradeMessage: string, incarnationNumber: number }} DisconnectionObject
+ * An Error-like object for use as the rejection reason of promises
+ * abandoned by upgrade.
+ *
+ * @typedef {{
+ *   name: string,
```
Comment (gibson042) [inline packages/internal/src/upgrade-api.js]: Should this be narrowed?
```suggestion
 *   name: 'vatUpgraded',
```

**Review by gibson042 — APPROVED** [review]
Thanks!

File: packages/SwingSet/src/types-external.js (hunk 1 above)
Comment (erights) [reply packages/SwingSet/src/types-external.js]: Gone

File: packages/internal/src/upgrade-api.js (hunk 2 above)
Comment (erights) [reply packages/internal/src/upgrade-api.js]: Good. Done.

**Pull request merged**